src/vm/amd64/cgenamd64.cpp

```diff
@@ -1126,6 +1126,121 @@ PCODE DynamicHelpers::CreateHelperWithTwoArgs(LoaderAllocator * pAllocator, TADD
     END_DYNAMIC_HELPER_EMIT();
 }
 
+//
+// Move an an argument into the second argument register and jump to a helper function.
+// Assumption: the first argument is already placed into the first argument register.
+//
+void DynamicHelpers::EmitJumpStubWithSecondArg(LoaderAllocator * pAllocator, BYTE*& pCode, PCODE pTarget, TADDR secondArgValue)
+{
+    CONTRACTL
+    {
+        GC_NOTRIGGER;
+        PRECONDITION(pCode != NULL && pTarget != NULL);
+    }
+    CONTRACTL_END;
+
+    // Move second argument into argument register (size = 10)
+#ifdef UNIX_AMD64_ABI
+    *(UINT16 *)pCode = 0xBE48; // mov rsi, XXXXXX
```

Comment:
`CreateHelperWithArg` has exact same code - call this method from there so that it is not duplicated.